src/main/java/org/openstreetmap/atlas/checks/validation/tag/HighwayAccessCheck.java

```diff
@@ -0,0 +1,113 @@
+package org.openstreetmap.atlas.checks.validation.tag;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import org.openstreetmap.atlas.checks.base.BaseCheck;
+import org.openstreetmap.atlas.checks.flag.CheckFlag;
+import org.openstreetmap.atlas.geography.atlas.items.AtlasObject;
+import org.openstreetmap.atlas.geography.atlas.items.Edge;
+import org.openstreetmap.atlas.geography.atlas.walker.OsmWayWalker;
+import org.openstreetmap.atlas.tags.AccessTag;
+import org.openstreetmap.atlas.tags.HighwayTag;
+import org.openstreetmap.atlas.utilities.configuration.Configuration;
+
+/**
+ * HighwayAccessCheck looks for ways that contain the access tag "yes" or "permissive". If the
+ * access tag is found, then the highway tag is also checked. Finally, the object is flagged if the
+ * highway tag is found in either motorway tags or in the footway tags provided in the beginning.
+ *
+ * @author v-naydinyan
+ */
+public class HighwayAccessCheck extends BaseCheck<Long>
+{
+
+    private static final long serialVersionUID = -5533238262833368666L;
+    private static final List<String> ACCESS_TAGS_TO_FLAG_DEFAULT = Arrays.asList("yes",
+            "permissive");
+    private static final List<String> HIGHWAY_TAGS_TO_FLAG_DEFAULT = Arrays.asList("motorway",
+            "trunk", "footway", "bridleway", "steps", "path", "cycleway", "pedestrian", "track",
+            "bus_guideway", "busway", "raceway");
+
+    private static final List<String> FALLBACK_INSTRUCTIONS = Arrays.asList(
+            "The access tag value is probably too generic for this way. A tag value of \"yes\" or \"permissive\" allows access to all types of traffic. If special access is granted or restricted on this way then please specify it or remove the access tag. See https://wiki.openstreetmap.org/wiki/Key:access?uselang=en#Transport_mode_restrictions for more information.");
```

Comment:
Should we be checking if there are already special restrictions paired with the access tag? I would guess we don't wan to flag features with something like access=yes and motor_car=no.

Reply:
This is what I found on OSM key access, "To avoid ambiguity, you may therefore want to avoid general tags access=yes and access=permissive, and use more specific transport modes where appropriate. For example, to distinguish a footway with open access from one with private access, use tags like foot=yes instead of access=yes." 
The assumption was that for each mode there shouldn't be an access=yes, unless it means that there is full access on the road. Hence, when we add the highway tag, we know that it is something specific. This is the logic that was used in the backend code, here is the link: https://github.com/osm-fr/osmose-backend/blob/c6c0e8a4598495c3b21c308081dbda92be0d1717/analysers/analyser_osmosis_highway_motorway.py

Reply:
Right, but just below that bit in the wiki it also has the following as a valid example:
`access=yes, motor_vehicle=no means that all transport modes except motor vehicles can use the element`

Reply:
I think the idea here is that we are combining both the access tag and the highway tag as described in this part, "In theory, adding access=yes to highway=footway could be read as changing default restrictions (which usually are foot=yes and vehicle=no for highway=footway) to yes, highway=footway + access=yes means "road, which is open for all pedestrians and vehicles". In practice, this combination is often used by mappers to modify (rather than enlarge) default values: for example, access=permissive with highway=steps is very unlikely to be traversable by a truck, whatever the tags may say." So, we are not just looking at access tag = yes but specifically a combination of that and the highway tag. Here is an Osmose example flagged for this check, where the tags are: [access:yes, highway:cycleway, motor_vehicle:no] https://www.openstreetmap.org/way/705630999 
I will ask around to see if this makes sense.

Reply:
Ah, right. That makes sense now. I did not fully think about the impact of the highway tag filter that is being used. Thanks for the clarification!